### PR TITLE
fix: multitenantnetwork reconciler should check errors correctly

### PR DESCRIPTION
**Pull request by thatmattlong**
**Reason for Change**:

Fixes a bug in multitenantreonciler that has been present since 1.4.13, preventing upgrade to latest CNS version. The bug is caused by comparing an error string with `strings.EqualFold`, where the error string is `response: UnknownContainerID` but the string comparison checks for `UnknownContainerID`. Since we have appropriately typed errors now, we can check the response code directly instead of doing a string comparison.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests

**Notes**:

### CODE REVIEW

File: cns/multitenantcontroller/mockclients/Makefile [hunk 1]
```diff
@@ -5,7 +5,7 @@ MOCKGEN = $(TOOLS_BIN_DIR)/mockgen
 
 generate: $(MOCKGEN) ## Generate mock clients
 	$(MOCKGEN) -source=$(REPO_ROOT)/cns/multitenantcontroller/multitenantoperator/multitenantcrdreconciler.go -package=mockclients cnsRESTservice > cnsrestservice_generated.go
-	$(MOCKGEN) -source=$(REPO_ROOT)/vendor/sigs.k8s.io/controller-runtime/pkg/client/interfaces.go -package=mockclients Client > kubeclient_generated.go
+	$(MOCKGEN) -source=$(REPO_ROOT)/vendor/sigs.k8s.io/controller-runtime/pkg/client/interfaces.go -package=mockclients Client,SubResourceWriter > kubeclient_generated.go
```
Comment (rbtr) [inline cns/multitenantcontroller/mockclients/Makefile:8]: can you provide some context on this?

Comment (rbtr) [inline cns/multitenantcontroller/mockclients/Makefile:8]: can you provide some context on this?

**Review by rbtr — DISMISSED** [review]

File: cns/multitenantcontroller/mockclients/Makefile (hunk 1 above)
Comment (thatmattlong) [reply cns/multitenantcontroller/mockclients/Makefile:8]: yes, previously our tests only used a mock of the `client.Client` but now we also need a mock of the `client.Status()`'s `client.SubResourceWriter`. So this is just being explicit about which symbols we want to generate.

Comment (rbtr) [reply cns/multitenantcontroller/mockclients/Makefile:8]: sounds good ty

File: cns/multitenantcontroller/multitenantoperator/multitenantcrdreconciler_test.go [hunk 2]
```diff
@@ -168,7 +232,7 @@ var _ = Describe("multiTenantCrdReconciler", func() {
 				NamespacedName: namespacedName,
 			})
 			Expect(err).NotTo(BeNil())
-			Expect(err.Error()).To(ContainSubstring("UnknownContainerID"))
```
Comment (neaggarwMS) [inline cns/multitenantcontroller/multitenantoperator/multitenantcrdreconciler_test.go:171]: oh wow we were validating the unexpected error code too

**Review by neaggarwMS — APPROVED** [review]

**Review by rbtr — APPROVED** [review]

**Pull request merged**